### ISSUE TITLE
typechecker/codegen: Inner type of None in a yield statement is not correctly outputted by codegen

### ISSUE THREAD

**Issue by cybergsus** [SerenityOS/jakt#952]
This program:

```jakt
/// Expect: selfhost-only
/// - output: "PASS\n"

function optional_foo() -> i32? => match true {
    false => {
        println("WHAT")
        yield 42i32
    }
    else => {
        yield None
    }
}



function main() {
    if not optional_foo().has_value() {
        println("PASS")
    }
}
```

Fails to put a type in the `yield None` from the `else`: 
```cpp
// [...]
        auto &&__jakt_enum_value = true;
            if (__jakt_enum_value == false) {
                return JaktInternal::ExplicitValue(({ Optional<i32> __jakt_var_0;           {
              outln(String("WHAT"));
              __jakt_var_0 = static_cast<i32>(42); goto __jakt_label_0;
          }
__jakt_label_0:; __jakt_var_0.release_value(); }));
            } else {
//              This should not be auto, but rather i32 ---------v
                return JaktInternal::ExplicitValue(({ Optional<auto> __jakt_var_1;           {
              __jakt_var_1 = JaktInternal::OptionalNone(); goto __jakt_label_1;
          }
// [...]
```

**Comment by jntrnr** [comment]
I think you can do:

```
    else => {
        let output: i32? = None
        yield output
    }
```
instead

**Comment by cybergsus** [comment]
Sure, I just put the issue in because there was a PR merged that seemed to work, so that it is known to not work in this case.

**Comment by lanmonster** [comment]
fixed as of 1ea81c6
